src/librustc/hir/mod.rs

```diff
@@ -2488,10 +2489,16 @@ pub type FreevarMap = NodeMap<Vec<Freevar<ast::NodeId>>>;
 
 pub type CaptureModeMap = NodeMap<CaptureClause>;
 
+pub type SmallHirIdVec = SmallVec<[HirId;1]>;
+pub type SmallNodeIdVec = SmallVec<[NodeId;1]>;
```

Comment:
I'd avoid introducing these new typedefs, they are not very useful.

Reply:
Would you use a constant for the length instead? It has to match across different modules.

Reply:
It's used only in a couple of places, simply `SmallVec<[HirId;1]>` would be good enough.

Also, style nit: arrays usually has a space after `;` - `[HirId; 1]`.

Reply:
By the way, have you measured how large these vectors actually are?
I'd expect the import chains with length 1 to be majority as well, but the reality turns out to be surprising sometimes.

Reply:
I've replaced the typedefs, but I can't tell how often the pattern occurs in real life. How would I measure that?

If this bug report is anything to go by, not that often: I mean, even if there are 5 duplicate reports, it hasn't been so frequent/annoying that anyone bothered to fix it yet.

I actually found this report because I thought I was affected, but it turns out it was because I was importing a trait that I was only using in a unit test, and so the import was indeed unused when compiling "for real". :-)